crates/walrus-service/src/event/event_blob_downloader.rs

```diff
@@ -140,33 +145,31 @@ impl EventBlobDownloader {
                 break;
             }
 
-            // Timer for waiting for the first certified event blob to be certified.
-            let start_time = Instant::now();
-            let blob_status = self
-                .read_event_blob_status(event_blob_id, reading_first_event_blob, start_time)
-                .await?;
-
-            if blob_status == BlobStatus::Nonexistent {
-                anyhow::ensure!(!blobs.is_empty(), "no available event blobs found");
-                tracing::info!(
-                    "stopping downloading event blobs with expired blob {}",
-                    event_blob_id
-                );
-                break;
-            }
-
             let blob_path = path.join(event_blob_id.to_string());
             let (blob, blob_source) = if blob_path.exists() {
                 (std::fs::read(blob_path.as_path())?, "local")
             } else {
-                let start_time = Instant::now();
-                self.read_event_blob(
-                    event_blob_id,
-                    blob_status,
-                    reading_first_event_blob,
-                    start_time,
-                )
-                .await?
+                // Timer for waiting for the first certified event blob to be certified.
+                let result = self
+                    .read_event_blob(event_blob_id, reading_first_event_blob)
```

Comment:
@sadhansood @mlegner after reading the client code a bit more, I think just using the `read_event_blob()` is cleaner, as internally, it'll try to first get the status. So we do one retry here for both status and blob tracking. PTAL